pydantic/_internal/_core_utils.py

```diff
@@ -83,7 +83,7 @@ def get_type_ref(type_: type[Any], args_override: tuple[type[Any], ...] | None =
     if isinstance(origin, TypeAliasType):
         type_ref = f'{module_name}.{origin.__name__}'
     else:
-        qualname = getattr(origin, '__qualname__', f'<No __qualname__: {origin}>')
+        qualname = getattr(origin, '__qualname__', '<No __qualname__>')
```

Comment:
```suggestion
        try:
            qualname = getattr(origin, '__qualname__', f'<No __qualname__: {origin}>')
        except Exception:
            qualname = getattr(origin, '__qualname__', f'<No __qualname__>')
```

Reply:
Yeah, this is what I mentioned on https://github.com/pydantic/pydantic/pull/6382#discussion_r1250725725

Reply:
(I've edited your msg because it had a f-string, fyi)